liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/BasicSupport.java

```diff
@@ -458,4 +472,26 @@ private InputStream getZipEntry(File zip, String entry) throws IOException {
         }
         return null;
     }
+
+    // Read WLP_OUTPUT_DIR from server.env. Return null if server.env doesn't
+    // exist or variable is not in server.env
+    private String getWlpOutputDir() throws IOException {
+        Properties envvars = new Properties();
+        
+        File serverEnvFile = new File(installDirectory, "etc/server.env");
+        if (serverEnvFile.exists()) {
+            envvars.load(new FileInputStream(serverEnvFile));
+        }
+        
+        if (serverEnv.exists()) {
+            envvars.load(new FileInputStream(serverEnv));
+        }
+        
+        serverEnvFile = new File(configDirectory, "server.env");
```

Comment:
How does this behave if configDirectory is null?

Reply:
If configDirectory is null then the new File instance is created as if by invoking the single-argument File constructor on the given child pathname string.